### ISSUE TITLE
Issue with screen and 256 colours

### ISSUE THREAD

**Issue by vilewrath** [busyloop/lolcat#117]
I have been trying to get lolcat to work with screen, and I've been having issues getting the colours to work correctly, attached is an image showing screen on the right, and no screen on the left. I used the script found [here](https://gist.github.com/hSATAC/1095100/) to generate the 256 colours that work at the top of the screen.

![image](https://user-images.githubusercontent.com/72516522/97019457-41298200-1548-11eb-8423-bfd29322b4b2.png)

Sorry if this isn't a lolcat issue, but any help would be greatly appreciated

**Comment by arnowi** [comment]
Hello [nyaalex](https://github.com/nyaalex),

Did you ever find a solution to this? Much appreciated.

**Comment by nyaalex** [comment]
No, but I did realise it wasn't an issue with lolcat, screen acts funny with 256 colour, I still don't know why, or how to fix it.

**Comment by arnowi** [comment]
I know it can work, it did on my previous work laptop. For some reason it doesn't on my new one despite copying over the same config/rc files as far as I know. It is indeed definitely a screen issue, not ZSH or Bash or Lolcat. Haven't tried with Tmux yet.

I'll post here if I do find a solution.

**Comment by biggianteye** [comment]
@nyaalex I don't think this problem is to do with 256-colour terminals. Your screenshot shows that your terminal is capable of showing all of those colours.

I have encountered this problem myself and have been looking into this and think it's down to an interaction between `lolcat`, `screen` and the `COLORTERM` variable. Terminals can advertise their ability to use 24-bit colour by setting `COLORTERM=truecolor`. `lolcat` has some specific handling for this which _might_ be the source of the problem.

Here is the an example using a new `screen` session in a terminal that sets `COLORTERM=truecolor` (iTerm). It doesn't render correctly in 24-bit mode but works when I disable that hint.

```shell
echo "[$TERM_PROGRAM] [$TERM] [$COLORTERM]"
date | LOLCAT_DEBUG=1 lolcat
date | COLORTERM= LOLCAT_DEBUG=1 lolcat
```

<img width="1005" alt="Terminal output after running the commands directly above this image" src="https://github.com/busyloop/lolcat/assets/1482649/527b117d-595f-4aa6-a572-bdc90e58dadb">


Here is the an example using a new `screen` session in a terminal that *doesn't* set `COLORTERM=truecolor` (macOS terminal). It renders correctly but fails when I enable the 24-bit mode hint.

```shell
echo "[$TERM_PROGRAM] [$TERM] [$COLORTERM]"
date | LOLCAT_DEBUG=1 lolcat
date | COLORTERM=truecolor LOLCAT_DEBUG=1 lolcat
```

<img width="1003" alt="Terminal output after running the commands directly above this image" src="https://github.com/busyloop/lolcat/assets/1482649/fe8033d6-df0a-4816-86fe-2b995511e0fc">

Could you please try out unsetting `COLORTERM` and see if that works for you?

I will continue to dig around the code to see if I can spot the problem. I'm not a ruby developer though, so I might just end up unsetting `COLORTERM` and leaving it there. We'll see.

**Comment by nyaalex** [comment]
Oh yeah, you absolutely are right, unsetting COLORTERM fixes it for me too. I said this was not a lolcat issue because this is common to almost every other application that tries to use 24-bit colour with screen, honestly I don't know why that perl script even works. Actually you have been extremely helpful because I never could figure out how to fix them, and unsetting COLORTERM seems like a fixall.

**Comment by biggianteye** [comment]
> Oh yeah, you absolutely are right, unsetting COLORTERM fixes it for me too

Great to hear!

> I said this was not a lolcat issue because this is common to almost every other application that tries to use 24-bit colour with screen

Can't say that I have knowingly used an application that has tried to use 24-bit colour before, but I'll keep an eye out. Having more examples will help with tracking down the source of the problem.

**Comment by ijin82** [comment]
@biggianteye thanks a lot! finally fixed

**Comment by iconoclasthero** [comment]
So the solution is:
`$ screen`
`$ unset COLORTERM`
remove `-t` flag from the invocation of `lolcat` for `whatever`
`$ whatever`

?
